test/unit/spec/common/simpleLogViewer-spec.js

```diff
@@ -0,0 +1,1096 @@
+                                    },
+                                },
+                            });
+                        } else if (args[0] === jcm.MESSAGE_TYPE.LOGS) {
+                            expectLogRequested(this);
+                            if (acc >= messageSeries.length) {
+                                // finish the test
+                                expect(this.simpleLogViewerInstance.model.getItem('lines')).toEqual(
+                                    logLines
+                                );
+                                return;
+                            }
+                            const messageParams = messageSeries[acc];
+                            expect(args[1]).toEqual({
+                                [jcm.PARAM.JOB_ID]: jobId,
+                                ...messageParams.request,
+                            });
+                            expect(this.simpleLogViewerInstance.model.getItem('lines')).toEqual(
+                                messageParams.model
+                            );
+                            testJobLogs(this, messageParams.model);
+                            TestUtil.sendBusMessage({
+                                bus: this.bus,
+                                type: jcm.MESSAGE_TYPE.LOGS,
+                                ...input[inputType],
+                                message: {
+                                    [jobId]: {
+                                        [jcm.PARAM.JOB_ID]: jobId,
+                                        latest: true,
+                                        batch_id: runningJob.batch_id,
+                                        ...messageParams,
+                                    },
+                                },
+                            });
+                            acc++;
+                        }
+                    });
+
+                    await this.simpleLogViewerInstance.start({
+                        node: this.node,
+                        jobId,
+                    });
+
+                    await TestUtil.waitForElementState(this.node, () => {
+                        return (
+                            this.node.querySelectorAll('.kb-log__log_line_container li').length >=
+                            logLines.length
+                        );
+                    });
+                    expect(this.simpleLogViewerInstance.model.getItem('lines')).toEqual(logLines);
+                    expectNotAwaitingLog(this);
+                });
+            });
+        });
+
+        describe('life cycle tests', () => {
+            beforeEach(function () {
+                createLogViewer(this, TEST_POLL_INTERVAL);
+            });
+            afterEach(() => {
+                TestUtil.clearRuntime();
+            });
+
+            ['job', 'batch'].forEach((inputType) => {
+                it(`has a life cycle when receiving status messages in ${inputType} mode`, async function () {
```

Comment:
mimic a job going through various states (queued, running, finished) and check that the widget displays appropriately.